### DISCUSSION TITLE
Response File support for MSVC

### DISCUSSION THREAD

**Discussion by temportalflux** [mozilla/sccache#1594] [General]
Hey folks, I'm looking to gather some information regarding the support for response files in sccache with msvc/msbuild. I'm a software engineer at Proletariat (a substudio of Blizzard Entertainment). We are using sccache for Mac and Linux, but with the command-line limitation on Windows, we need to use response-files when compiling with msvc/msbuild. SCCache doesn't currently support this feature and we'd like to use sccache on windows if we can make it work without maintaining a long-term fork.

I've done some digging and as I currently understand it:

- #107 mentions an interest in supporting response files for msvc
- The above discussion inspired #192 in 2017, but it was abandoned by the author after it took ~6 months for feedback/approval
- That PR was followed by another PR (#378), which was also abandoned after the author was redirected to #192
- #1082 was opened in Dec 2021, which again requests support for response files in msvc, but has not received any replies. This issue is still open.
- #1183 repeated the same request in June 2022, and a comment was added referencing all of the above tickets. This issue is still open.

We have a working implementation inspired by #192 in an internal fork of sccache, based off of latest main / 0.4.0, but would rather not maintain a fork in the long term. Given the history of this feature request, we wanted to reach out and see if there is interest in collaborating before opening a new PR.

So my actual questions here are:

- Are folks receptive to/interested in receiving a new PR for responses files in msvc?
- If so, do folks have bandwidth to review and provide feedback on a new PR, with the goal of getting it merged for an upcoming release?
- Provided there is interest and bandwidth, what kind of timeline can we (author and reviewer(s)) commit to that is sustainable for folks involved? We want to get a solid understanding of what to expect so we can plan accordingly internally.

We're still getting final approval from our legal team, but hope that in the mean time we can identify some of these answers with y'all in the hopes that we can collaborate in a sustainable timeframe for all of us!

**Comment by sylvestre** [comment]
> Are folks receptive to/interested in receiving a new PR for responses files in msvc?

Sure, as long as they are coming with tests, docs, etc 

For the rest of your questions, it depends on the quality of the PR. ;)

**Reply by temportalflux** [reply]
Are the docs you're referring to the ones under the `/docs` folder in the repo?

**Reply by sylvestre** [reply]
yes

**Reply by sylvestre** [reply]
and in the code :)

**Reply by temportalflux** [reply]
Great! My current plan is to make a doc for Response Files (which will cover both msvc and gcc), plus thoroughly documented code/changes. Are there any other docs that you'd expect to see?

**Comment by temportalflux** [comment]
Pull Request opened https://github.com/mozilla/sccache/pull/1618
